spec/integration/product-manager.spec.js

```diff
@@ -0,0 +1,145 @@
+import _ from 'lodash'
```

Comment:
Let's not create integration tests for every class. They should really be blackbox integration testing where you call the main method of Variant reassignment and then you check the result of the whole reassignment process. So it means we will have only integration tests created from the wiki, which should cover all the methods we have. For the services, you can create unit classes, it should be enough.

Reply:
Hello @lojzatran please read my opinion on service integration tests here https://github.com/commercetools/commercetools-node-variant-reassignment/pull/9#discussion_r154577140

With integration tests only for variant reassignment process, you won't cover a lot of different edge cases in service methods. Also, you will test a lot of parts repeatedly (eg. fetching products...) just to test different behavior in one method so the tests will do a lot of useless steps which takes time. I created this issue in which we should remove redundant tests https://github.com/commercetools/commercetools-node-variant-reassignment/issues/16

Reply:
I think we're covering all edge cases in our wiki. Or do you think there are cases that we didn't cover? Let's leave the test here but maybe we don't need more integration tests.

Reply:
I think that in our examples we did not cover a lot of possible cases. But .. it does not make sense to test them in examples because we would make a lot of integration tests which are doing the same steps just with few small differences (like removing one variant or removing one variant and promoting another to masterVariant).

This can be tested with higher efficiency in productManager integration/unit tests.

Also the more functions we add to productManager (eg: anonymizing variants will be the next) the more combinations you have so you should test rather separated methods instead of the whole process for all possible use cases. It will be much faster (you won't test same methods over and over) and much easier for future changes (when you change one method you will not have to change it in all tests). Also, the tests will be small because they target just small pieces of the code instead of big tests which tests whole process.